reactor-core/src/main/java/reactor/core/publisher/FluxRetryWhen.java

```diff
@@ -47,41 +47,56 @@
 	static final Duration MAX_BACKOFF = Duration.ofMillis(Long.MAX_VALUE);
 
 	final Function<? super Flux<Throwable>, ? extends Publisher<?>> whenSourceFactory;
+	final boolean resetOnNext;
 
 	FluxRetryWhen(Flux<? extends T> source,
 							  Function<? super Flux<Throwable>, ? extends Publisher<?>> whenSourceFactory) {
+		this(source, whenSourceFactory, false);
+	}
+
+	FluxRetryWhen(Flux<? extends T> source,
+							  Function<? super Flux<Throwable>, ? extends Publisher<?>> whenSourceFactory,
+			boolean resetOnNext) {
 		super(source);
 		this.whenSourceFactory = Objects.requireNonNull(whenSourceFactory, "whenSourceFactory");
+		this.resetOnNext = resetOnNext;
 	}
 
 	static <T> void subscribe(CoreSubscriber<? super T> s, Function<? super
-			Flux<Throwable>, ?
-			extends Publisher<?>> whenSourceFactory, CorePublisher<? extends T> source) {
+			Flux<Throwable>, ? extends Publisher<?>> whenSourceFactory, CorePublisher<? extends T> source,
+			boolean resetOnNext) {
 		RetryWhenOtherSubscriber other = new RetryWhenOtherSubscriber();
 		Subscriber<Throwable> signaller = Operators.serialize(other.completionSignal);
 
 		signaller.onSubscribe(Operators.emptySubscription());
 
 		CoreSubscriber<T> serial = Operators.serialize(s);
 
-		RetryWhenMainSubscriber<T> main =
-				new RetryWhenMainSubscriber<>(serial, signaller, source);
+		Runnable otherReset = () -> {
```

Comment:
**Allocation police** (lol 😂):
WDYT about avoiding this `Runnable` and introducing a more specialised `RetryWhenMainSubscriber` OR passing a boolean (and other necessary variables) to the existing one

Reply:
I'll look into passing the arguments directly to the ctor rather than relying on the `Runnable` capturing that state. That said allocation wise it is only one allocation per _original_ subscription (not on the critical data path).

Reply:
I initially started with passing these elements, but it was cumbersome